### DISCUSSION TITLE
How to update ui.plot?

### DISCUSSION THREAD

**Discussion by korenmolcho** [zauberzeug/nicegui#55] [Q&A] [answered]
Hello,

I'm trying to create an interface where data can be loaded by user selection and plotted.
I've managed to change the variables holding the data and the labels that indicate new data has been loaded.
When it comes to the plot, I can't seem to update it, I set "close"  to False, and still, I can't change the initial plot.

please help I'm desperate
here's an example:

```python3
from nicegui import ui
import matplotlib.pyplot as plt

data = {'C': 20, 'C++': 15, 'Java': 30,
        'Python': 35}

with ui.plot(figsize=(10, 10), close=False) as plot:
    ax = plot.fig.add_subplot()
    # dig, ax = plt.subplots(figsize=(10, 10))
    courses = list(data.keys())
    values = list(data.values())
    ax.barh(courses, values, color='maroon')

def changeChart():
    ax.set_ylabel('abc')
    ui.update()

ui.button('Button', on_click=changeChart)

ui.run(host="127.0.0.1", port=8000)
```

**Comment by falkoschindler** [comment, marked as answer]
You need to change two things:

1. Modify the axes within a `with plot:` context.
2. Don't forget to add `plot` as an argument to `ui.update`.

```python3
def changeChart():
    with plot:
        ax.set_ylabel('abc')
    ui.update(plot)
```

I have to admit: That's not very intuitive. At least step 2 could be done automatically when leaving the plot context. We might change that in a future release.

**Reply by korenmolcho** [reply]
Amazing! 
Thanks a lot!
